documentation/asciidoc/computers/raspberry-pi/bootflow-2711.adoc

```diff
@@ -73,3 +73,16 @@ Please see the xref:raspberry-pi.adoc#raspberry-pi-4-bootloader-configuration[bo
 === Bootloader Updates
 
 The bootloader may also be updated before the firmware is started if a `pieeprom.upd` file is found. Please see the xref:raspberry-pi.adoc#raspberry-pi-4-boot-eeprom[bootloader EEPROM] page for more information about bootloader updates.
+
+=== Fail-safe OS updates (TRYBOOT)
+
+The bootloader/firmware provide a one-shot flag which, if set, is cleared but causes 'tryboot.txt' to be loaded instead of 'config.txt'. This alternate firmware would specify the pending OS update firmware, cmdline, kernel and OS-prefix files. Since the flag is cleared before starting the firwmare a crash or reset will cause the original 'config.txt' file to be loaded on the next reboot.
+
+To set the `tryboot` flag add `tryboot` after the partition number in the 'reboot' command.
+```
+# Quotes are important. Reboot only accepts a single argument.
+sudo reboot `0 tryboot`
+```
+
+`tryboot` is supported on all Raspberry Pi models, however, on Pi 4 model B revision 1.0 and 1.1 the EEPROM must not be write protected. This is because older Raspberry Pi 4B devices have to reset the power supply (losing the tryboot state) so this is stored inside the EEPROM instead.
```

Comment:
"supported on all Raspberry Pi models" - including models before the Pi 4? If that's the case, then perhaps `bootflow-2711.adoc` isn't the best place for this? :shrug:

Reply:
ping @timg236 :wink: 
Or is this "all Raspberry Pi models based on the BCM2711" ?

Reply:
It was added for BCM2711 but does actually work on Pi1..Pi3

Reply:
If it's not Pi4-specific, then perhaps https://github.com/raspberrypi/documentation/blob/develop/documentation/asciidoc/computers/raspberry-pi/bootflow-legacy.adoc might be a better place? :shrug:  (although that page says "BCM2837 only", which is slightly awkward :confused: )
Maybe https://github.com/raspberrypi/documentation/blob/develop/documentation/asciidoc/computers/configuration/boot_folder.adoc makes sense? (as I guess that's where `tryboot.txt` would need to be located)

Reply:
Let's just get this out there, it can be moved elsewhere if needed